library-best-practices/JLBP-18.md

```diff
@@ -0,0 +1,50 @@
+[JLBP-18] Only shade dependencies as a last resort
+--------------------------------------------------
+
+Shading is a process where selected classes (generally from dependencies)
+from the classpath of a library are relocated to different Java packages and
```

Comment:
"the classpath of a library" --> a library's dependencies

since libraries technically don't have classpaths. Only runtime environments have classpaths.

Reply:
Done